blockchain/proxy_test.go

```diff
@@ -1203,3 +1241,231 @@ func TestProxy_IsDataTrieMigrated(t *testing.T) {
+			"https://test.org/block/by-nonce/21000000": blockResponseBytes,
+			"https://test.org/block/by-hash/00f7d0e806c5ff3700236f78c67007c8000000000000000000409a07ff835d8e": blockResponseBytes,
+		}
+
+		httpClient := createMockClientMultiResponse(responseMap)
+		args := createMockArgsProxy(httpClient)
+		ep, _ := NewProxy(args)
+
+		res, err := ep.FilterLogs(context.Background(), validFilter)
+
+		assert.Nil(t, err)
+		assert.Equal(t, len(res), 1)
+		assert.Equal(t, res[0].Identifier, "ESDTTransfer")
+	})
+
+	t.Run("should work without caching", func(t *testing.T) {
+		validFilter := &sdkCore.FilterQuery{
+			BlockHash: &[32]byte{
+				0x00, 0xf7, 0xd0, 0xe8, 0x06, 0xc5, 0xff, 0x37,
+				0x00, 0x23, 0x6f, 0x78, 0xc6, 0x70, 0x07, 0xc8,
+				0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+				0x00, 0x40, 0x9a, 0x07, 0xff, 0x83, 0x5d, 0x8e,
+			},
+			FromBlock: core.OptionalUint64{21000000, false},
+			ToBlock:   core.OptionalUint64{21000000, false},
+			ShardID:   0,
+			Topics:    nil,
+		}
+
+		blockResponseBytes := []byte(httpDataBlock21000000)
+		statusResponseBytes := []byte(httpNodeStatus)
+
+		responseMap := map[string][]byte{
+			"https://test.org/node/status":             statusResponseBytes,
+			"https://test.org/block/by-nonce/21000000": blockResponseBytes,
+			"https://test.org/block/by-hash/00f7d0e806c5ff3700236f78c67007c8000000000000000000409a07ff835d8e": blockResponseBytes,
+		}
+
+		httpClient := createMockClientMultiResponse(responseMap)
+		args := createMockArgsProxy(httpClient)
+		ep, _ := NewProxy(args)
+
+		res, err := ep.FilterLogs(context.Background(), validFilter)
+
+		assert.Nil(t, err)
+		assert.Equal(t, len(res), 4)
+		assert.Equal(t, res[0].Identifier, "ESDTTransfer")
+		assert.Equal(t, res[1].Identifier, "writeLog")
+		assert.Equal(t, res[2].Identifier, "completedTxEvent")
+		assert.Equal(t, res[3].Identifier, "completedTxEvent")
+	})
+
+	t.Run("should work with caching", func(t *testing.T) {
+
+		validFilter := &sdkCore.FilterQuery{
+			FromBlock: core.OptionalUint64{21000001, true},
+			ToBlock:   core.OptionalUint64{21000001, true},
+			ShardID:   0,
+			Topics:    nil,
+			BlockHash: nil,
+		}
+
+		blockResponseBytes := []byte(httpDataBlock21000001)
+		statusResponseBytes := []byte(httpNodeStatus)
+
+		responseMap := map[string][]byte{
+			"https://test.org/node/status":             statusResponseBytes,
+			"https://test.org/block/by-nonce/21000001": blockResponseBytes,
+			"https://test.org/block/by-hash/00f7d0e806c5ff3700236f78c67007c8000000000000000000409a07ff835d8e": blockResponseBytes,
+		}
+
+		httpClient := createMockClientMultiResponse(responseMap)
+		args := createMockArgsProxyWithCache(httpClient)
+		ep, _ := NewProxy(args)
+
+		_, err := ep.FilterLogs(context.Background(), validFilter)
+		assert.Nil(t, err)
+
+		// check if the block is cached
+		cacheKey := make([]byte, 8)
+		binary.BigEndian.PutUint64(cacheKey, 21000001)
+		value, _ := ep.filterQueryBlockCacher.Get(cacheKey)
```

Comment:
this checked should be carried directly on the args.FilterQueryBlockCacher
Alternatively, a stub implementation could have signaled through some bool variables that the put was called.